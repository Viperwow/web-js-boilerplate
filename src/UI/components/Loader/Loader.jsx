import {identity as _identity} from 'lodash';

const Loader = ({
  text,
  t = _identity,
}) => t(text);

export default Loader;
