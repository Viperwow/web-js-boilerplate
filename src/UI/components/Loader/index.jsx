import {constant as _constant} from 'lodash';

const Loader = _constant('Loading...');

export default Loader;
