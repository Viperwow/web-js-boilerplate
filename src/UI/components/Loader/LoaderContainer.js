import withTranslate from 'src/common/withTranslate';
import connectQuery from 'src/common/connectQuery';
import {compose} from 'src/helpers/utility';
import {GET_APP_LOCALE} from 'src/queries';
import Loader from './Loader';

const mapStateToProps = () => ({
  text: 'components.loader',
});

export default compose(
  withTranslate,
  connectQuery(GET_APP_LOCALE, mapStateToProps),
)(Loader);
