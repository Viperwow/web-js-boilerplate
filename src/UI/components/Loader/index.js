import withTranslate from 'src/common/withTranslate';
import connectQuery from 'src/common/connectQuery';
import {compose} from 'src/helpers/utility';
import {LOCALE_QUERY} from 'src/queries';
import Loader from './Loader';

const mapStateToProps = (_, {t}) => ({
  text: t('components.loader'),
});

export default compose(
  withTranslate,
  connectQuery(LOCALE_QUERY, mapStateToProps),
)(Loader);
