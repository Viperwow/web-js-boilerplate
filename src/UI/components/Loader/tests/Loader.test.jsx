import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import Loader from '../Loader';

jest.unmock('../Loader');

describe('Loader component should render correctly', () => {
  test('With default props', () => {
    const output = shallow(<Loader />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });

  test('With text prop', () => {
    const output = shallow(<Loader text="text" />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });
});
