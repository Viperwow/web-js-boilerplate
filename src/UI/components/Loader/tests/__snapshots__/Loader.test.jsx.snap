// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render correctly With default props 1`] = `""`;

exports[`Loader component should render correctly With text prop 1`] = `"text"`;
